v5_ws_private.go

```diff
@@ -1,10 +1,256 @@
+	UpdatedTime     string     `json:"updatedTime"`
+	TpslMode        TpSlMode   `json:"tpslMode"`
+	LiqPrice        string     `json:"liqPrice"`
+	BustPrice       string     `json:"bustPrice"`
+	Category        CategoryV5 `json:"category"`
+	PositionStatus  string     `json:"positionStatus"`
+}
+
+// Key :
+func (r *V5WebsocketPrivatePositionResponseContent) Key() V5WebsocketPrivateParamKey {
+	return V5WebsocketPrivateParamKey{
+		Topic: r.Topic,
+	}
+}
+
+// addParamPositionFunc :
+func (s *V5WebsocketPrivateService) addParamPositionFunc(param V5WebsocketPrivateParamKey, f func(V5WebsocketPrivatePositionResponseContent) error) error {
+	if _, exist := s.paramPrivateMap[param]; exist {
+		return errors.New("already registered for this param")
+	}
+	s.paramPrivateMap[param] = f
+	return nil
+}
+
+// retrievePositionFunc :
+func (s *V5WebsocketPrivateService) retrievePositionFunc(key V5WebsocketPrivateParamKey) (func(V5WebsocketPrivatePositionResponseContent) error, error) {
+	f, exist := s.paramPrivateMap[key]
+	if !exist {
+		return nil, errors.New("func not found")
+	}
+	return f, nil
+}
+
+// judgeTopic :
+func (s *V5WebsocketPrivateService) judgeTopic(respBody []byte) (V5WebsocketPrivateTopic, error) {
+	parsedData := map[string]interface{}{}
+	if err := json.Unmarshal(respBody, &parsedData); err == nil {
+		if topic, ok := parsedData["topic"].(string); ok {
+			return V5WebsocketPrivateTopic(topic), nil
+		}
+		if authStatus, ok := parsedData["success"].(bool); ok {
+			if !authStatus {
+				return "", errors.New("auth failed: " + parsedData["ret_msg"].(string))
+			}
+		}
+		return "", nil
+	} else {
+		return "", err
+	}
+}
+
+// parseResponse :
+func (s *V5WebsocketPrivateService) parseResponse(respBody []byte, response interface{}) error {
+	if err := json.Unmarshal(respBody, &response); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Subscribe :
+func (s *V5WebsocketPrivateService) Subscribe() error {
+	param, err := s.client.buildAuthParam()
+	if err != nil {
+		return err
+	}
+	if err := s.connection.WriteMessage(websocket.TextMessage, param); err != nil {
+		return err
+	}
+	return nil
+}
+
+// RegisterFuncPosition :
+func (s *V5WebsocketPrivateService) RegisterFuncPosition(f func(V5WebsocketPrivatePositionResponseContent) error) error {
+	key := V5WebsocketPrivateParamKey{
+		Topic: V5WebsocketPrivateTopicPosition,
+	}
+	if err := s.addParamPositionFunc(key, f); err != nil {
+		return err
+	}
+	param := struct {
+		Op   string        `json:"op"`
+		Args []interface{} `json:"args"`
+	}{
+		Op:   "subscribe",
+		Args: []interface{}{V5WebsocketPrivateTopicPosition},
+	}
+	buf, err := json.Marshal(param)
+	if err != nil {
+		return err
+	}
+	if err := s.connection.WriteMessage(websocket.TextMessage, buf); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Start :
+type ErrHandler func(isWebsocketClosed bool, err error)
+
+func (s *V5WebsocketPrivateService) Start(errHandler ErrHandler, ctx context.Context) error {
```

Comment:
nits:

context.Context should be the first parameter of a function

Reply:
Right, it's been merged now, but I'll implement the Order websocket soon, so will take care of this in that upcoming PR

Reply:
nevermind, I see you added a commit